app/src/androidTest/java/com/android/sample/ui/activityDetails/CommentItemTest.kt

```diff
@@ -163,20 +174,49 @@ class CommentItemTest {
           comments = listOf(testComment), // Pass in the comment with its reply
           onReplyComment = { _, _ -> },
           onDeleteComment = {},
-          onAddComment = {})
+          onAddComment = {},
+          creatorId = "1")
     }
 
     // Ensure the original comment is displayed
-    composeTestRule.onNodeWithText("Amine: This is a comment").assertIsDisplayed()
+    composeTestRule.onNodeWithText("This is a comment").assertIsDisplayed()
+    composeTestRule.onNodeWithText("Amine:")
 
     // Ensure the reply is displayed
-    composeTestRule.onNodeWithText("John: This is a reply").assertIsDisplayed()
+    composeTestRule.onNodeWithText("This is a reply").assertIsDisplayed()
+    composeTestRule.onNodeWithText("John:").assertIsDisplayed()
 
     // Check that the reply is well nested, ensuring its indented and positioned under its origin
     // comment
     composeTestRule
-        .onNodeWithText("John: This is a reply")
+        .onNodeWithText("This is a reply")
         .assertIsDisplayed()
         .assertLeftPositionInRootIsEqualTo(40.05.dp)
   }
```

Comment:
ensure that the CommentItem correctly displays a badge for the creator of the comment and does not display the badge for non-creators. This enhances the UI to visually differentiate comments made by the creator from those made by other users.